pkg/sql/codegen/xgboost/codegen.go

```diff
@@ -17,42 +17,33 @@ import (
 	"bytes"
 	"encoding/json"
 	"fmt"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/attribute"
 	"strings"
 
 	"sqlflow.org/sqlflow/pkg/sql/codegen"
 )
 
-var attributeChecker = map[string]func(interface{}) error{
-	"eta": func(x interface{}) error {
-		switch x.(type) {
-		case float32, float64:
-			return nil
-		default:
-			return fmt.Errorf("eta should be of type float, received %T", x)
-		}
-	},
-	"num_class": func(x interface{}) error {
-		switch x.(type) {
-		case int, int32, int64:
-			return nil
-		default:
-			return fmt.Errorf("num_class should be of type int, received %T", x)
-		}
-	},
-	"train.num_boost_round": func(x interface{}) error {
-		switch x.(type) {
-		case int, int32, int64:
-			return nil
-		default:
-			return fmt.Errorf("train.num_boost_round should be of type int, received %T", x)
-		}
-	},
-	"objective": func(x interface{}) error {
-		if _, ok := x.(string); !ok {
-			return fmt.Errorf("objective should be of type string, received %T", x)
-		}
-		return nil
-	},
+func newFloat32(f float32) *float32 {
+	return &f
```

Comment:
I am not sure if returning the address of a parameter is to *new* a float value. Maybe 

```go
return &new(float32)
```

Reply:
Go supports returning address of stack variables: https://stackoverflow.com/questions/52996452/returning-pointer-from-a-local-variable-in-function